crates/geometry/src/convex_hull.rs

```diff
@@ -1,56 +1,44 @@
 use linear_algebra::Point2;
-use nalgebra::Matrix2;
+use parry2d::transformation::convex_hull;
 
-pub enum Range {
-    Full,
-    OnlyBottomHalf,
-}
-
-pub fn reduce_to_convex_hull<Frame>(points: &[Point2<Frame>], range: Range) -> Vec<Point2<Frame>> {
-    // https://en.wikipedia.org/wiki/Gift_wrapping_algorithm
-    // Modification: This implementation iterates from left to right until a smaller x value is found
-    if points.is_empty() {
-        return vec![];
-    }
-    let mut point_on_hull = *points
-        .iter()
-        .min_by(|a, b| a.x().total_cmp(&b.x()))
-        .unwrap();
-    let mut convex_hull = vec![];
-    loop {
-        convex_hull.push(point_on_hull);
-        let mut candidate_end_point = points[0];
-        for point in points.iter() {
-            let last_point_on_hull_to_candidate_end_point = candidate_end_point - point_on_hull;
-            let last_point_on_hull_to_point = *point - point_on_hull;
-            let determinant = Matrix2::from_columns(&[
-                last_point_on_hull_to_candidate_end_point.inner,
-                last_point_on_hull_to_point.inner,
-            ])
-            .determinant();
-            let point_is_left_of_candidate_end_point = determinant < 0.0;
-            if candidate_end_point == point_on_hull || point_is_left_of_candidate_end_point {
-                candidate_end_point = *point;
-            }
-        }
-        // begin of modification
-        if matches!(range, Range::OnlyBottomHalf) && candidate_end_point.x() < point_on_hull.x() {
-            break;
-        }
-        // end of modification
-        point_on_hull = candidate_end_point;
-        if candidate_end_point == *convex_hull.first().unwrap() {
-            break;
-        }
+pub fn reduce_to_convex_hull<Frame>(points: &[Point2<Frame>]) -> Vec<Point2<Frame>> {
+    if points.len() < 3 {
+        return points.to_vec();
     }
-    convex_hull
+
+    // SAFETY: linear_algebra::Point2 has repr(transparent) and is guaranteed to have the same memory layout as nalgebra::Point2
+    let points: &[_] = unsafe { std::mem::transmute(points) };
```

Comment:
use `transmute` above:


```suggestion
    let points: &[_] = unsafe { transmute(points) };
```